src/PubSub/Subscription.php

```diff
@@ -0,0 +1,579 @@
+            return false;
+        }
+    }
+
+    /**
+     * Get info on a subscription
+     *
+     * If the info is already cached on the object, it will return that result.
+     * To fetch a fresh result, use {@see Subscription::reload()}.
+     *
+     * Example:
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $info = $subscription->info();
+     * echo $info['name']; // `projects/my-awesome-project/subscriptions/my-new-subscription`
+     * ```
+     *
+     * @param  array $options Configuration options
+     * @return array Subscription data
+     */
+    public function info(array $options = [])
+    {
+        if (!$this->info) {
+            $this->reload($options);
+        }
+
+        return $this->info;
+    }
+
+    /**
+     * Retrieve info on a subscription from the API.
+     *
+     * To use the previously cached result (if it exists), use
+     * {@see Subscription::info()}.
+     *
+     * Example:
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $subscription->reload();
+     * $info = $subscription->info();
+     * echo $info['name']; // `projects/my-awesome-project/subscriptions/my-new-subscription`
+     * ```
+     *
+     * @param  array $options Configuration options
+     * @return array Subscription data
+     */
+    public function reload(array $options = [])
+    {
+        return $this->info = $this->connection->getSubscription($options + [
+            'subscription' => $this->name
+        ]);
+    }
+
+    /**
+     * Retrieve new messages from the topic.
+     *
+     * Example:
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $messages = $subscription->pull();
+     * foreach ($messages as $message) {
+     *      echo $message['data'];
+     * }
+     * ```
+     *
+     * @param  array $options {
+     *      Configuration Options
+     *
+     *      @type bool $returnImmediately If set, the system will respond
+     *            immediately, even if no messages are available. Otherwise,
+     *            wait until new messages are available.
+     *      @type int  $maxMessages Limit the amount of messages pulled.
+     * }
+     * @return \Generator
+     */
+    public function pull(array $options = [])
+    {
+        $options['pageToken'] = null;
```

Comment:
Dave and I have talked some about how to implement manual paging. We're not sure yet how we want it to work, so right now, yes, nextPageToken is hidden.